lm_eval/tasks/gsm.py

```diff
@@ -0,0 +1,206 @@
+"""PAL: Program-aided Language Models
+https://arxiv.org/abs/2211.10435
+
+GSM-8k: Training Verifiers to Solve Math Word Problems
+https://arxiv.org/abs/2110.14168
+
+In PaL, Large Language Model solves reasoning problems that involve complex arithmetic and procedural tasks by generating 
+reasoning chains of text and code.This offloads the execution of the code to a program runtime, in our case, a Python interpreter.
+
+This task implements PAL methodology to evaluate GSM-8k and GSM-Hard benchmarks.
+"""
+
+import os, re, json
+from evaluate import load
+from lm_eval.base import Task
+from typing import Union
+from lm_eval.tasks.custom_metrics.pal_code_exec import compute
+from enum import Enum
+
+
+_CITATION = """
+@article{gao2022pal,
+  title={PAL: Program-aided Language Models},
+  author={Gao, Luyu and Madaan, Aman and Zhou, Shuyan and Alon, Uri and Liu, Pengfei and Yang, Yiming and Callan, Jamie and Neubig, Graham},
+  journal={arXiv preprint arXiv:2211.10435},
+  year={2022}
+}
+
+@article{cobbe2021gsm8k,
+  title={Training Verifiers to Solve Math Word Problems},
+  author={Cobbe, Karl and Kosaraju, Vineet and Bavarian, Mohammad and Chen, Mark and Jun, Heewoo and Kaiser, Lukasz and Plappert, Matthias and Tworek, Jerry and Hilton, Jacob and Nakano, Reiichiro and Hesse, Christopher and Schulman, John},
+  journal={arXiv preprint arXiv:2110.14168},
+  year={2021}
+}
+"""
+# Number of few shot examples to consider
+NUM_SHOTS = 2
```

Comment:
If 8 few-shot examples fit in the 2048 context window (including room for generation), it's better to keep it to have comparable results, otherwise we can keep less. The two-shot setting started with APPS few-shot setting which had very long examples and we kept that number, but it doesn't have to be the case here.

We can even add an argument to the evaluation harness to specify the number of few-shot examples to include  and select them randomly from the test set if they're not provided like in [lm-evaluation-harness](https://github.com/EleutherAI/lm-evaluation-harness/blob/3d14707a1054e174553d7e4ec99d1cb8905735bb/lm_eval/evaluator.py#LL37C12-L37C23) and [here](https://github.com/EleutherAI/lm-evaluation-harness/blob/3d14707a1054e174553d7e4ec99d1cb8905735bb/lm_eval/base.py#L576). But this could be the subject of another PR.

Reply:
Thanks for the suggestion. 8 shot prompt token length comes around to 1400, the remaining 600 tokens should be more than enough room for generations. Typical solution generated seem to take 80-100 tokens. So, I have updated the shot count to 8.

I agree that it would be good to have shot count as a argument that can be controlled by the user while running the script.